File: 3. Scripts/ruby/pa-email-already-used2.rb
require 'selenium-webdriver'
require 'rspec'
require_relative "singup_page.rb"
require_relative "user_page.rb"

email = "test@test.me"
already_used_emailtext = "Uneta E-mail adresa je već zauzeta"
password = "testic123"

# Test fir registration at polovni automoboili
describe 'Polovni automobili application:' do
  describe 'signup to the polovni automobili application' do
    it 'Confirm user already used this email address' do
      @driver = Selenium::WebDriver.for :firefox
      # Go to sing-in form
      @driver.navigate.to 'https://www.polovniautomobili.com/registracija'

      # Fill out form
      signup = SignupPage.new(@driver)
      signup.enter_email_address(email)
      signup.enter_password(password)
      
      sleep 1

      # Confirm user already used this email address
      user = UserPage.new(@driver)
      already_used_email = user.user_already_used_email
      expect(already_used_email).to eq(already_used_emailtext)
      
      @driver.quit

    end
  end
end

File: 3. Scripts/javascript/pa-email-already-used2.js
const { Builder, By } = require('selenium-webdriver');
const assert = require('assert');
const SignupPage = require('./signup_page');
const UserPage = require('./user_page');

const email = 'test@test.me';
const alreadyUsedEmailText = 'Uneta E-mail adresa je već zauzeta';
const password = 'testic123';

describe('Polovni automobili application:', function() {
  describe('signup to the polovni automobili application', function() {
    it('Confirm user already used this email address', async function() {
      const driver = await new Builder().forBrowser('firefox').build();

      // Go to sign-in form
      await driver.get('https://www.polovniautomobili.com/registracija');

      // Fill out form
      const signup = new SignupPage(driver);
      await signup.enterEmailAddress(email);
      await signup.enterPassword(password);

      await driver.sleep(1000);

      // Confirm user already used this email address
      const user = new UserPage(driver);
      const alreadyUsedEmail = await user.userAlreadyUsedEmail();
      assert.strictEqual(alreadyUsedEmail, alreadyUsedEmailText);

      await driver.quit();
    });
  });
});